### PR TITLE
Implement PostgreSQL broker

### PR DESCRIPTION
**Pull request by mook-as**
Mostly based on how the MySQL one was done.

To use:

1. Export `DOCKER_ORGANIZATION` and `DOCKER_REPOSITORY` as desired.
1. `make all build-image publish-image helm`
1. `helm install ./output/helm/` with most of the variables from the CF deployment; the unusual parameters are:
- `--namespace pg` makes your life easier
- `SERVICE_LOCATION=http://cf-usb-sidecar-postgres.pg.svc.cluster.local:8081` (see namespace above)
- `SERVICE_POSTGRESQL_HOST=AUTO` will deploy a Postgres server along with the broker bits; you can also specify an existing server (in which case give the password too)
- `SERVICE_POSTGRESQL_SSLMODE=disable` is probably necessary to actually connect to it

### CODE REVIEW

File: csm-extensions/services/dev-postgres/Dockerfile-setup [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM cf-usb-sidecar-buildbase:latest
+
+COPY chart/scf-connector.sh /usr/local/bin/
+COPY chart/authorize_ca.sh  /usr/local/bin/
+WORKDIR /
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y unzip
```
Comment (alexjh) [inline csm-extensions/services/dev-postgres/Dockerfile-setup]: Should have an `apt-get clean` at the end of these commands.

**Comment by alexjh** [comment]
@mook-as, from the instructions above:

> make all build-image publish-image helm

There's no helm mentioned in the Makefile. Typo?

**Comment by mook-as** [comment]
It's in the [postgres one](https://github.com/SUSE/cf-usb-sidecar/pull/3/files#diff-e52ee17318b5d1edbdd2ac909c62f643R146). You only need to `make all build-image` at the top level, IIRC.

File: csm-extensions/services/dev-postgres/Dockerfile-setup (hunk 1 above)
Comment (mook-as) [reply csm-extensions/services/dev-postgres/Dockerfile-setup]: Fixed by multi-stage building (so we just throw away the whole container).

File: csm-extensions/services/dev-postgres/chart/values.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+env:
+  CF_ADMIN_PASSWORD: ~
+  CF_ADMIN_USER: ~
+  CF_CA_CERT: ~
+  CF_DOMAIN: ~
+  SERVICE_LOCATION: ~
+  SERVICE_POSTGRESQL_HOST: ~
+  SERVICE_POSTGRESQL_PASS: cf-usb-sidecar-mysql-admin-password
```
Comment (alexjh) [inline csm-extensions/services/dev-postgres/chart/values.yaml]: This refers to `mysql`? Is this intentional? If so, a comment here would clarify why.

Comment (mook-as) [reply csm-extensions/services/dev-postgres/chart/values.yaml]: No, just find-and-replace failure. Fixed in 6d1a3aa5cdc71814d70ddd85aa7e9355ea748f06.

File: scripts/dev-extensions/build-docker-image.sh [hunk 3]
```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
```
Comment (alexjh) [inline scripts/dev-extensions/build-docker-image.sh]: This could use a `set -e`.

**Pull request merged**